crates/uv/src/cli.rs

```diff
@@ -1955,6 +1957,30 @@ pub(crate) struct ToolRunArgs {
     pub(crate) python: Option<String>,
 }
 
+#[derive(Args)]
+pub(crate) struct ToolchainNamespace {
+    #[command(subcommand)]
+    pub(crate) command: ToolchainCommand,
+}
+
+#[derive(Subcommand)]
+pub(crate) enum ToolchainCommand {
+    /// List the available toolchains.
+    List(ToolchainListArgs),
+}
+
+#[derive(Args)]
+#[allow(clippy::struct_excessive_bools)]
+pub(crate) struct ToolchainListArgs {
+    /// List all available toolchains, including those that do not match the current platform.
+    #[arg(long)]
+    pub(crate) all: bool,
+
+    /// Only list installed toolchains.
+    #[arg(long)]
+    pub(crate) only_installed: bool,
```

Comment:
No attachment to these defaults yet, probably going to change this interface as we understand the uses of listing toolchains.

Probably going to add support for listing all the system toolchains we can find too.

Reply:
Maybe by default we should only list the latest patch version for the downloads.

Reply:
Should these be marked as conflicting?

Reply:
Good call yeah they should be